pixeltable/catalog/table_version.py

```diff
@@ -593,9 +613,21 @@ def add_columns(
         self.version += 1
         preceding_schema_version = self.schema_version
         self.schema_version = self.version
-        status = self._add_columns(cols, print_stats=print_stats, on_error=on_error)
+        index_cols: dict[Column, tuple[index.IndexBase, Column, Column]] = {}
+        all_cols: list[Column] = []
         for col in cols:
-            _ = self._add_default_index(col)
+            all_cols.append(col)
```

Comment:
goal is to call add_columns with all the columns that we will create for default index, this way if there is any error/interruption _add_columns can cleanup correctly